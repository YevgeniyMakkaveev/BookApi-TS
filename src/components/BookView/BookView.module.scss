@import '../var';

.card {
 margin: 0 auto;
 padding: 2rem;

 .category {
  padding: 1rem 0;
  color: $color_gray;
  font-weight: 600;
 }

 .mainInfo {
  display: grid;
  grid-template-columns: 0.8fr 1.2fr;

  img {
   display: block;
   align-self: center;
   min-width: 200px;
   max-width: 300px;
  }

  .infoList {
   text-align: left;

   .title {
    font-size: 2rem;
    font-size: 700;
   }

   .list {
    list-style: none;
    text-align: left;
   }
  }

  .descriptionWrap {
   margin-top: 2rem;
   grid-column-start: 1;
   grid-column-end: 3;
   align-items: center;
   padding: 0 auto;
   border: double $color_brown;

   .description {
    margin: 2rem 0;
   }
  }
 }
}

@media screen and (min-width: $breakpoint-tablet) {
 .card {
  font-size: 20px;

  .mainInfo {
   img {
    min-width: 300px;
    max-width: 400px;
   }

   .infoList {
    .title {
     font-size: 2.3rem;
     font-size: 700;
    }
    .list{
     li{
      margin: 0.5rem 0;
     }
    }
   }
  }
 }
}