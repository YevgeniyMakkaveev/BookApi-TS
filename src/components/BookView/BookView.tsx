import React from "react";
import IBookView from "../../types/bookView";
import getImage from "../../units/getImage";
import classes from "./BookView.module.scss";

const BookView: React.FC<IBookView> = (props) => {
  const {
    authors,
    categories,
    description,
    imageLinks,
    publishedDate,
    publisher,
    title,
    industryIdentifiers,
    pageCount,
  } = props;
  const mainText = description ? (
    <div
      dangerouslySetInnerHTML={{ __html: description }}
      className={classes.description}
    />
  ) : (
    <div className={description}>Description of the book is not found.</div>
  );
  return (
    <div className={classes.card}>
      <div className={classes.category}>
        {categories ? categories.join() : "No categories"}
      </div>

      <div className={classes.mainInfo}>
        <img src={getImage(imageLinks, false)} alt={title} />
        <div className={classes.infoList}>
          <h2 className={classes.title}> {title}</h2>
          <ul className={classes.list}>
            <li>
              <strong>Authors: </strong>
              {authors ? authors.join(" ") : "No authors"}
            </li>
            <li>
              <strong>Publisher:</strong>
              {publisher}
            </li>
            <li>
              <strong>Published date:</strong> {publishedDate}
            </li>
            <li>
              <strong>Pages:</strong> {pageCount}
            </li>
            <li>
              <strong>ID:</strong>{" "}
              {industryIdentifiers
                ? industryIdentifiers[0].type +
                  " " +
                  industryIdentifiers[0].identifier
                : "No ID"}
            </li>
          </ul>
        </div>
        <div className={classes.descriptionWrap}>{mainText}</div>
      </div>
    </div>
  );
};
export default BookView;
