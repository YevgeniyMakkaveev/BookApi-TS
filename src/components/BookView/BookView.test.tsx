import React from 'react';
import { render, screen } from '@testing-library/react';

import BookView from './BookView';

const book =  <BookView authors={["Jon","Doe"]} categories={["Any","category"]} description="awersome book" imageLinks={ {smallThumbnail: "No", thumbnail:"image"} } industryIdentifiers={[{identifier: "No", type: "id"}]} pageCount={300} publishedDate={"1984"} publisher={"Me"} title={"Book"} />

describe('Search component works', () => {
 it('should render correctly', ()=>{
render(book)
    expect(screen.getByText("Book")).toBeInTheDocument();
    expect(screen.getByText("Any,category")).toBeInTheDocument();
    expect(screen.getByText("Jon Doe")).toBeInTheDocument();
    expect(screen.getByText("Me")).toBeInTheDocument();

 })
})
 describe("it should be used to test snapshots", () => {

  it("should mach the snapshot", () =>{
   render(book)
 expect(book).toMatchSnapshot();
  })
 })