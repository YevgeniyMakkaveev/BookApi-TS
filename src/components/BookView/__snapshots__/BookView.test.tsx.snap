// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should be used to test snapshots should mach the snapshot 1`] = `
<BookView
  authors={
    Array [
      "Jon",
      "Doe",
    ]
  }
  categories={
    Array [
      "Any",
      "category",
    ]
  }
  description="awersome book"
  imageLinks={
    Object {
      "smallThumbnail": "No",
      "thumbnail": "image",
    }
  }
  industryIdentifiers={
    Array [
      Object {
        "identifier": "No",
        "type": "id",
      },
    ]
  }
  pageCount={300}
  publishedDate="1984"
  publisher="Me"
  title="Book"
/>
`;
